src/opm/output/eclipse/WriteInit.cpp

```diff
@@ -423,79 +423,41 @@ namespace {
 
     template <typename T, class WriteVector>
     void writeCellPropertiesWithDefaultFlag(const Properties& propList,
-                                            const ::Opm::GridProperties<T>&
-#ifndef ENABLE_3DPROPS_TESTING
-                                            propValues
-#endif
-                                            , const ::Opm::FieldPropsManager& fp,
-                                            const ::Opm::EclipseGrid&
-#ifndef ENABLE_3DPROPS_TESTING
-                                            grid
-#endif
-                                            , WriteVector&&                   write)
+                                            const ::Opm::FieldPropsManager& fp,
+                                            WriteVector&&  write)
     {
         for (const auto& prop : propList) {
-#ifdef ENABLE_3DPROPS_TESTING
             if (! fp.has<T>(prop.name))
                 continue;
 
             auto data = fp.get<T>(prop.name);
             auto defaulted = fp.defaulted<T>(prop.name);
             write(prop, std::move(defaulted), std::move(data));
-#else
-            if (! propValues.hasKeyword(prop.name)) {
-                continue;
-            }
-
-            const auto& opm_property = propValues.getKeyword(prop.name);
-            const auto& dflt         = opm_property.wasDefaulted();
-
-            write(prop, grid.compressedVector(dflt),
-                  opm_property.compressedCopy(grid));
-#endif
         }
     }
 
     template <typename T, class WriteVector>
     void writeCellPropertiesValuesOnly(const Properties& propList,
-                                       const ::Opm::GridProperties<T>&
```

Comment:
This function is now only used for floating-point properties so you can remove the template argument `T` and use `has<double>()` and `get<double>()` in the function body.